modules/nodes-pods-autoscaling-custom-prometheus-config.adoc

```diff
@@ -0,0 +1,175 @@
++
+[source,terminal]
+----
+$ oc describe serviceaccount <service_account>
+----
++
+--
+where:
+
+<service_account>:: Specifies the name of the service account.
+--
++
+.Example output
+[source,terminal]
+----
+Name:                thanos
+Namespace:           my-project
+Labels:              <none>
+Annotations:         <none>
+Image pull secrets:  thanos-dockercfg-nnwgj
+Mountable secrets:   thanos-dockercfg-nnwgj
+Tokens:              thanos-token-9g4n5 <1>
+Events:              <none>
+
+----
+<1> Use this token in the trigger authentication.
+
+. Create a trigger authentication with the service account token:
+
+.. Create a YAML file similar to the following:
++
+[source,yaml]
+----
+apiVersion: keda.sh/v1alpha1
+kind: TriggerAuthentication
+metadata:
+  name: keda-trigger-auth-prometheus
+spec:
+  secretTargetRef: <1>
+  - parameter: bearerToken <2>
+    name: thanos-token-9g4n5 <3>
+    key: token <4>
+  - parameter: ca
+    name: thanos-token-9g4n5
+    key: ca.crt
+----
+<1> Specifies that this object uses a secret for authorization.
+<2> Specifies the authentication parameter to supply by using the token.
+<3> Specifies the name of the token to use.
+<4> Specifies the key in the token to use with the specified parameter.
+
+.. Create the CR object:
++ 
+[source,terminal]
+----
+$ oc create -f <file-name>.yaml
+----
+
+. Create a role for reading Thanos metrics:
++
+.. Create a YAML file with the following parameters:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: thanos-metrics-reader
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - get
+- apiGroups:
+  - metrics.k8s.io
+  resources:
+  - pods
+  - nodes
+  verbs:
+  - get
+  - list
+  - watch
+----
+
+.. Create the CR object:
++ 
+[source,terminal]
+----
+$ oc create -f <file-name>.yaml
+----
+
+. Create a role binding for reading Thanos metrics:
++
+.. Create a YAML file similar to the following:
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: thanos-metrics-reader <1>
+  namespace: my-project <2>
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: thanos-metrics-reader
+subjects:
+- kind: ServiceAccount
+  name: thanos <3>
+  namespace: my-project <4>
+----
+<1> Specifies the name of the role you created.
+<2> Specifies the namespace of the object you want to scale.
+<3> Specifies the name of the service account to bind to the role.
+<4> Specifies the namespace of the object you want to scale.
+.. Create the CR object:
++ 
+[source,terminal]
+----
+$ oc create -f <file-name>.yaml
+----
+
+You can now deploy a scaled object or scaled job to enable autoscaling for your application, as described in the following sections. To use {product-title} monitoring as the source, in the trigger, or scaler, specify the `prometheus` type and use `\https://thanos-querier.openshift-monitoring.svc.cluster.local:9092` as the `serverAddress`.
```

Comment:
Would be nice if we can show here the actual ScaledObject and highlight that the user needs to use `\https://thanos-querier.openshift-monitoring.svc.cluster.local:9092` as the `serverAddress` and that `namespace` property in the trigger configuration must be set to the namespace where the workload is deployed.
https://github.com/zroubalik/keda-openshift-examples/blob/main/prometheus/ocp-monitoring/scaledobject.yaml